src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php

```diff
@@ -153,9 +173,18 @@ protected function printPendingMessagesMessage(ReceiverInterface $receiver, Symf
         }
     }
 
-    protected function getReceiver(): ReceiverInterface
+    protected function getReceiver(/* string $name */): ReceiverInterface
     {
-        return $this->receiver;
+        if (1 > \func_num_args() && __CLASS__ !== static::class && __CLASS__ !== (new \ReflectionMethod($this, __FUNCTION__))->getDeclaringClass()->getName() && !$this instanceof \PHPUnit\Framework\MockObject\MockObject && !$this instanceof \Prophecy\Prophecy\ProphecySubjectInterface) {
```

Comment:
What was the problem that made you need to add the extra logic logic related to the test environment? That's definitely not going to work - but there may be some tricky issue that someone smarter (than me) will know the solution to.